vncviewer/EmulateMB.h

```diff
@@ -21,29 +21,50 @@
 
 #include <rfb/Timer.h>
 #include <rfb/Rect.h>
+#include <rdr/types.h>
 
 class EmulateMB : public rfb::Timer::Callback {
 public:
-  EmulateMB();
+  EmulateMB(rdr::U32 emulateMBModKey);
 
   void filterPointerEvent(const rfb::Point& pos, int buttonMask);
 
+  bool filterKeyPress(int keyCode, rdr::U32 keySym);
+  bool filterKeyRelease(int keyCode);
```

Comment:
You probably need a reset function as well. Unfortunately you aren't guaranteed to get a release event for every press. So we need a way to reset the state machine when focus is lost.